Control/lib/services/QCConfiguration.service.js

```diff
@@ -79,10 +83,22 @@ class QCConfigurationService {
     return parsedData;
   }
 
+  /**
+   * Get configuration restrictions by key from Consul
+   * @param {string} key - the key of the configuration
+   * @return {Promise<TypeMap>}
+   * @throws {NotFoundError | ServiceUnavailableError}
```

Comment:
Being a promise it will not 'throw' but reject. Above on L51 and below L101, you added the Error type to the promise. 
It would be good to keep one convention only.